docs(config): rewrite VS Code debugging section (#22974)
vitejs/vite: docs/config/index.md

})
```

## Debugging the Config File in VS Code

For the most reliable debugging experience, use the native config loader when starting Vite:

```bash
vite --configLoader native
```

The native loader executes the original config file directly, so breakpoints in the config file and in plugin hooks such as `transform` map to the original source. It requires a runtime that supports the syntax used by your config file, such as Node.js 22.18+ for TypeScript files.

> When using `--configLoader bundle` (the current default, though `native` is planned to become the default in a future major version), Vite generates an inline source map and writes the bundled config to `node_modules/.vite-temp` before loading it. If you need to use the bundled loader, add the temporary directory to the source map locations used by the VS Code JavaScript Debug Terminal:

Reviewer: ```suggestion
When using `--configLoader bundle` (the current default, though `native` is planned to become the default in a future major version), Vite generates an inline source map and writes the bundled config to `node_modules/.vite-temp` before loading it. If you need to use the bundle loader, add the temporary directory for the JavaScript Debug Terminal in `.vscode/settings.json`:
```

I think we should keep the file name so they know where to put this. Also I shorten it a little so it's a bit less wordy and following up with the paragraph below.

Author: Thanks! 👍

  }
}
```

This setting only applies to the JavaScript Debug Terminal; it does not affect launch configurations started from the Run and Debug view.

> If you start Vite from the Run and Debug view instead of a JavaScript Debug Terminal, put `resolveSourceMapLocations` in the launch configuration:

Reviewer: ```suggestion
This setting only applies to the JavaScript Debug Terminal, it does not affect launch configurations started from the Run and Debug view. To support this for the Run and Debug view, add the temporary directory in `.vscode/launch.json`:
```

Also shorten here a bit.

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Vite",

>       "runtimeExecutable": "pnpm",

Reviewer: I think we should use `npm` though since it's the default most people use
```suggestion
      "runtimeExecutable": "npm",
```
